core/src/desul/atomics/SYCLConversions.hpp

```diff
@@ -9,50 +9,92 @@ SPDX-License-Identifier: (BSD-3-Clause)
 #ifndef DESUL_ATOMICS_SYCL_CONVERSIONS_HPP_
 #define DESUL_ATOMICS_SYCL_CONVERSIONS_HPP_
 #ifdef DESUL_HAVE_SYCL_ATOMICS
-#include "desul/atomics/Common.hpp"
 #include <CL/sycl.hpp>
 
+#include "desul/atomics/Common.hpp"
+
 namespace desul {
+namespace Impl {
+
+#ifdef __clang__
+namespace sycl_sync_and_atomics = ::sycl::ONEAPI;
```

Comment:
This will need to be updated after #4244 (when we update our CI to a newer compiler version).